### DISCUSSION TITLE
Monaco editor fails to resolve in my test

### DISCUSSION THREAD

**Discussion by BrendonLapp** [vitest-dev/vitest#1806] [Q&A]
Hi, I've been implementing vitest into our vite vue app recently and have ran into issues related to the Monaco Editor. I keep running into the following error:
`Error: Failed to resolve entry for package "monaco-editor". The package may have incorrect main/module/exports specified in its package.json. `
and I'm not sure how to fix it? At this point I'm pretty confused on where to go from this? I Know there are a couple other answers about the monaco editor about but I can't seem to get the solutions there to work for me?

This is what my vitest.config.ts file looks like:
```
import { defineConfig } from "vitest/config";
import vue from "@vitejs/plugin-vue";

export default defineConfig({
	plugins: [vue()],
	test: {
		globals: true,
		environment: "jsdom",
	},
});
```

And I saw one post that said to add this:
```
deps: {
      inline: ['monaco-editor']
    }
```
But that also doesn't work for me what it's added to the `test: {}`.

How would I be able to resolve this?

I'm trying to do a simple test like the following:
```
import Login from "../pages/login.vue";
import { mount } from "@vue/test-utils";

describe("Login Page", () => {
	const wrapper = mount(Login);

	it("mounts the component", () => {
		expect(wrapper).toBeTruthy();
	});
});
```

**Comment by xavirovira** [comment]
I solved it by adding this alias inside the "test" object of defineConfig() in the vite.config file

 ```
alias: [
      {
        find: /^monaco-editor$/,
        replacement:
          __dirname + "/node_modules/monaco-editor/esm/vs/editor/editor.api",
      },
    ],
```

**Reply by j-a-m-l** [reply]
Thanks a lot @xavirovira for writing your solution: it has helped me a lot.

**Comment by KurtGokhan** [comment]
I think this might have been solved in the newest version of `monaco-editor`. I tried with `0.55.1` and it seems to work without requiring the workaround above.
